src/utils/openmc/tallies.py

```diff
@@ -1443,19 +835,61 @@ def __init__(self):
         self._tallies_file = ET.Element("tallies")
 
 
-    def add_tally(self, tally):
+    def add_tally(self, tally, merge=False):
 
         if not isinstance(tally, Tally):
             msg = 'Unable to add a non-Tally {0} to the TalliesFile'.format(tally)
             raise ValueError(msg)
 
-        self._tallies.append(tally)
+        if merge:
+            merged = False
+
+            # Look for a tally to merge with this one
+            for i, tally2 in enumerate(self._tallies):
+
+                # If a mergeable tally is found
+                if tally2.can_merge(tally):
+
+                    # Replace tally 2 with the merged tally
+                    merged_tally = tally2.merge(tally)
+                    self._tallies[i] = merged_tally
+                    merged = True
```

Comment:
Since there's no `break` here, isn't is possible that if there are multiple potential tallies to merge with, the new tally would get merged with all of them?